src/Compilation.zig

```diff
@@ -1297,10 +1297,10 @@ test "addSourceFromReader" {
             defer comp.deinit();
 
             var buf_reader = std.io.fixedBufferStream(str);
-            const source = try comp.addSourceFromReader(buf_reader.reader(), "path", @intCast(u32, str.len));
+            const source = try comp.addSourceFromReader(buf_reader.reader(), "path", @intCast(str.len));
 
             try std.testing.expectEqualStrings(expected, source.buf);
-            try std.testing.expectEqual(warning_count, @intCast(u32, comp.diag.list.items.len));
+            try std.testing.expectEqual(warning_count, @as(u32, @intCast(comp.diag.list.items.len)));
```

Comment:
Shouldn't this get the type from `warning_count`?

Reply:
It looks like it's not able to see through the `@TypeOf(expected)` where `expected` is `anytype` - without the cast I get:

```
src/Compilation.zig:1303:56: error: @intCast must have a known result type
            try std.testing.expectEqual(warning_count, @intCast(comp.diag.list.items.len));
                                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/Compilation.zig:1303:56: note: result type is unknown due to anytype parameter
src/Compilation.zig:1303:56: note: use @as to provide explicit result type
```

Reply:
That seems like a bug.